collections/_posts/2023-06-11-DALLE2.md

```diff
@@ -0,0 +1,127 @@
+
+The output of the decoder is of size 64x64 and it takes two upsampling models to get to the final resolution (64x64 -> 256x256 and 256x256 -> 1024x1024).
+
+An important part is that the image generation is not (only) conditionned by the caption but it uses the authors found that using CLIP embeddings gave better results. Examples of results are shown later in the review.
+
+* CLIP model uses a ViT-H/16 image encoder with 256x256 images, a width of 1280 and 32 Transformer blocks.
+* The CLIP text encoder is a Transformer with causal attention mask, width of 1024 and 24 transformer blocks.
+* Encoder is trained with data from CLIP and DALL-E dataset (~ 650M images)
+* Prior, decoder and upsamplers are only trained with the DALL-E dataset (~250M images)
+
+> Even if unCLIP is called 'DALL-E 2', the way it works is very different from the first version of DALL-E (based on VQ-VAE).
+
+&nbsp;
+
+# Prior
+
+The authors have trying two different models for the prior, an autoregressive model, and a diffusion models. Both showed similar results but the latter was more computationnally efficient so it is the one that was kept (the details of ow the AR was used will thus not be detailled here).
+
+A diffusion model aims to generate data from a target distribution from pure Gaussian noise. The process can be seen as a Markov chain during which at each step, the data is corrupted by a small amount of noise. A diffusion model is trained to reverse this corrupting process (see this [tutorial](https://creatis-myriad.github.io/tutorials/2022-06-20-tutorial_transformer.html) for more detailled explanations).
+
+* The diffusion model used here is a decoder's Transformer[^1] with causal attention mask, width of 1664 and 24 blocks. 
+* The sequence of tokens used is the following : the encoded text $$y$$, the CLIP text embedding $$z_t$$, an embedding for the diffusion timestep, the noised CLIP image embedding whose output at the end of the diffusion process will then be used in the decoder.
+* Training objective function : $$ L_{prior} = \mathbb{E}_{t \sim [1,T], z_i^{(t)} \sim q_t} [ \lVert f_{\theta} (z_i^{(t)}, t, y) - z_i \rVert_2^2 ]$$.
+* To improve sampling quality, two samples of $$z_i$$ are predicted and only the one with the higher dot product with $$z_t$$ is kept.
+ 
+> Note that the model aims to predict directly the unnoised $$z_i$$ and not the noise like it is most commonly done in diffusion models.
+
+# Decoder
+
+The decoder is greatly inspired from GLIDE[^2], a previous text-guided generative **diffusion model** developed by OpenIA.
+
+* The diffusion is a DDIM, not a standard DDIM. It has two main features : allowing faster sampling without loss of quality and it has an additional parameter $$\eta$$ such that the generation is deterministic when $$\eta=0$$ and stochastic when $$\eta > 0$$ (and it is equivalent to DDPM when $$\eta=1$$)
```

Comment:
Is the first acronym correct here? Because saying that the *diffusion is a DDIM, not a standard DDIM* does not intuitively make sense to me? So I though maybe you meant something else here.

Reply:
thanks, it is ''not the standard DDPM" (+ I added the full name behind the acronym DDIM and a reference to the paper)